### PR TITLE
Issue-2063 - Add support for countries to require state

### PR DESCRIPTION
**Pull request by AleksandrsKondratjevs**
Extended magento module to provide country with boolean if it requires to have state.

### CODE REVIEW

**Comment by alfredsgenkins** [comment]
Hey! Why is this module named `directory-graphql` ? (ok I got it, divered from M2 module name)

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+    "name": "scandipwa/directory-graphql",
+    "description": "Magneto 2 Directory related functionality extension",
+    "type": "magento2-module",
+    "require": {
+        "php": "~7.3.0||~7.4.0",
+        "../../../../vendor/magento/module-directory": "100.4.*",
+        "../../../../vendor/magento/module-store": "101.1.*",
+        "../../../../vendor/magento/module-graph-ql": "100.4.*",
+        "../../../../vendor/magento/framework": "103.0.*"
```
Comment (alfredsgenkins) [inline composer.json]: This is not the way to specify dependencies, use real names instead.

File: src/Model/Resolver/Countries.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        $countries = $this->countryInformationAcquirer->getCountriesInfo();
+        $countriesRequireState = $this->directoryHelper->getCountriesWithStatesRequired();
+        $output = [];
+
+        // Added check for country if it requires state/province
+        foreach ($countries as $country) {
+            $result = $this->dataProcessor->buildOutputDataArray($country, CountryInformationInterface::class);
+
+            if (in_array($result['id'], $countriesRequireState)) {
+                $result['is_state_required'] = (boolean) true;
```
Comment (alfredsgenkins) [inline src/Model/Resolver/Countries.php]: You are setting it to boolean value already, why cast here?

**Review by alfredsgenkins — CHANGES_REQUESTED** [review]

File: src/Model/Resolver/Countries.php (hunk 2 above)
Comment (AleksandrsKondratjevs) [reply src/Model/Resolver/Countries.php]: Hey! Yea, it is not needed here. For some reason when at first was setting without cast I was getting null in response, perhaps it was just cache issue.

**Pull request merged**